src/main/java/cloud/eppo/ConfigurationStore.java

```diff
@@ -20,9 +20,20 @@ public class ConfigurationStore {
   private Map<String, BanditParameters> banditParameters;
 
   public ConfigurationStore() {
+    this(null, null);
+  }
+
+  public ConfigurationStore(String initialFlagConfiguration, String initialBanditParameters) {
     flags = new ConcurrentHashMap<>();
     banditReferences = new ConcurrentHashMap<>();
     banditParameters = new ConcurrentHashMap<>();
+
+    if (initialFlagConfiguration != null) {
+      setFlagsFromJsonString(initialFlagConfiguration);
```

Comment:
minor(unrelated to this PR): `ConcurrentHashMap` is used incorrectly in `ConfigurationStore` as they are never modified — only read. It's better to use a regular hashmap here.

major(but again unrelated to the PR): `flags`/`banditReferences` fields are overwritten without any synchronization, so users may see inconsistent values (e.g., bandit references out-of-sync with flags, etc) or with inconsistent ordering.

Reply:
I've moved to a `ConfigurationBuffer` object to encapsulate the both flag and bandit data and the decoding of the config. This moves setting the configuration to a single operation. I've also switched to immutable maps since we're not modifying them.